main.go

```diff
@@ -69,8 +69,9 @@ func convertJunitToSlack(junitFiles ...*junit.Suites) []slack.Attachment {
 			}
 
 			for _, result := range suite.Results {
+				failure := result.Failure
```

Comment:
It would be easier to read as a separate `resultToAttachment()`

Reply:
Separated out most of the logic below this to a separate function. Hopefully, it's a bit easier to read now. Feel free to add more suggestions regarding the readability